core/hash/dig_spec.rb

```diff
@@ -0,0 +1,40 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is '2.3' do
+  describe "Hash#dig" do
+
+    it "returns [] with one arg" do
+      h = { 0 => false, :a => 1 }
+      h.dig(:a).should == 1
+      h.dig(0).should be_false
+      h.dig(1).should be_nil
+    end
+
+    it "does recurse" do
+      h = { foo: { bar: { baz: 1 } } }
+      h.dig(:foo, :bar, :baz).should == 1
+      h.dig(:foo, :baz).should be_nil
```

Comment:
I would add here an extra assertion like:
` h.dig(:foo, :bar, :nope).should be_nil` since implementations seem highly biased towards the one-arg case :smile: